packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3863,9 +3926,14 @@ async function setFakeProvider(
     stubLookupNetworkWhileSetting?: boolean;
     stubGetEIP1559CompatibilityWhileSetting?: boolean;
   } = {},
-): Promise<ProviderEngine> {
-  const fakeMetamaskProvider = buildFakeMetamaskProvider(stubs);
-  createMetamaskProviderMock.mockReturnValue(fakeMetamaskProvider);
+): Promise<SafeEventEmitterProvider> {
```

Comment:
Following the comments elsewhere, I feel like this  shouldn't return a promise of SafeEventEmitterProvider, but a promise of FakeProvider.